src/Umbraco.PublishedCache.HybridCache/Persistence/DatabaseCacheRepository.cs

```diff
@@ -277,19 +285,25 @@ public async Task<IEnumerable<ContentCacheNode>> GetMediaSourcesAsync(IEnumerabl
 
     private async Task OnRepositoryRefreshed(IContentCacheDataSerializer serializer, ContentCacheNode content, bool preview)
     {
-
         ContentNuDto dto = GetDtoFromCacheNode(content, !preview, serializer);
 
-        await Database.InsertOrUpdateAsync(
-            dto,
-            "SET data = @data, dataRaw = @dataRaw, rv = rv + 1 WHERE nodeId = @id AND published = @published",
-            new
-            {
-                dataRaw = dto.RawData ?? Array.Empty<byte>(),
-                data = dto.Data,
-                id = dto.NodeId,
-                published = dto.Published,
-            });
+        dto.RawData ??= [];
+        dto.Rv++;
+        ContentNuDto? existing = await Database.FirstOrDefaultAsync<ContentNuDto>(
+            Sql()
+                .SelectAll()
+                .From<ContentNuDto>()
+                .Where<ContentNuDto>(x => x.NodeId == dto.NodeId && x.Published == dto.Published));
+        if (existing is null)
+        {
+            _ = await Database.InsertAsync(dto);
+        }
+        else
+        {
+            Sql<ISqlContext> updateSql = Sql().Update<ContentNuDto>(u => u.Set(d => d.Data, dto.Data).Set(rd => rd.RawData, dto.RawData).Set(v => v.Rv, dto.Rv))
+                .Where<ContentNuDto>(x => x.NodeId == dto.NodeId && x.Published == dto.Published);
+            _ = await Database.ExecuteAsync(updateSql);
+        }
```

Comment:
I'm not really too keen on this update, as it by-passes the helper we have for `InsertOrUpdateAsync`.

Firstly, why is is necessary to do this and not just use the helper?  I assume you are looking to replace the SQL statement for the `SET`, but this looks to be standard SQL - is it a problem for particular database providers?

If it is necessary to change, could you look to amend the `InsertOrUpdateAsync` helper itself?  That way we'd fix all instances of it's usage, and not have to apply the more complicated code you've added here in potentially other places.  Perhaps an overload taking a `Sql` parameter instead of a `string` for the `SET` statement could work.

Reply:
Indeed I hesitated to do this change. I have tested this with all four local databases: SQL Server 2019/2022, SQLite and PostgreSQL 17 and it worked. I haven't test Azure SQL Server what is of course very important for Umbraco Cloud.
So if you feal better with a minor change quoting the necessary column names I am fine with that.
Maybe this could be reconsidered in a future release...